### DISCUSSION TITLE
Hide user chat controls

### DISCUSSION THREAD

**Discussion by aimana007** [open-webui/open-webui#4950] [Q&A]
Is there an easy way to hide the "Chat Controls" for the "user" level? I'd prefer that users not be able to fine tune models that I have worked on, or to override the system prompt.

**Comment by shadowfaxproject** [comment]
This will be super helpful feature in cases of building consumer facing apps that do not require users (in this case consumers) that level of control.

**Reply by aimana007** [reply]
Not only in a commercial case, but for example, I do not want someone who is using it (locally) as a tutor to break the system prompts out. I have it set to not discuss certain topics, and I do not want that overridden by curious/smart kids.

**Comment by Thinkpiet** [comment]
I don't think there is an easy way like an environment variable or another configuration option. However, if you install manually from the Github repo, then you can go to `src/lib/components/layout/Navbar.svelte` and remove these lines:
```
<Tooltip content={$i18n.t('Controls')}>
	<button
		class=" flex cursor-pointer px-2 py-2 rounded-xl hover:bg-gray-50 dark:hover:bg-gray-850 transition"
		on:click={() => {
			showControls = !showControls;
		}}
	>
		<div class=" m-auto self-center">
			<AdjustmentsHorizontal className=" size-5" strokeWidth="0.5" />
		</div>
	</button>
</Tooltip>
```
Run`npm run build` again. That will hide the  "Chat Controls" button.

**Reply by aimana007** [reply]
I'm running the docker image because it's just for a local user or two so that's not easily possible (it's in some JS file buried in the /build/_app/immutable/ directory somewhere inside the container instance). I guess it wouldn't be too bad to just switch to the github node source build... but this might be a good feature request because it seems like it could be useful for others as well.

**Comment by rotemdan** [comment]
The latest version has the chat controls as a widget that is docked to the right.

The problem with that is that it takes space where the scrollbar should be, making access to the scrollbar require a kind of "pixel-hunting", rather than then just positioning the mouse at the rightmost edge of the viewport.

I found this workaround to remove the widget. It uses the [Stylus](https://chromewebstore.google.com/detail/stylus/clngdbkpkpeebahjckkjfobafhncgmne) extension to apply a custom stylesheet to the page:

```css
@-moz-document domain("localhost") {
    div[data-pane-group] > div[role="separator"] {
        display: none;
    }
}
```

Even though it works, it's not fully ideal since it will apply to any website or web application run at localhost, and would hide anything that matches the pattern.

Using a more specific selector like `div[data-pane-id="lTOoPbf2Ik"] > div[role="separator"]` may not be reliable since it would likely break the moment the ID is changed (this can happen even on a minor version update).

**Edit**: it also apparently causes the chat controls widget to become permanently hidden, even if the chat control button is pressed.

**Comment by basjaburg** [comment]
Has there any solution to this already?

**Reply by Classic298** [reply]
This option has already been implemented with a new permission toggle

**Reply by mkagit** [reply]
> This option has already been implemented with a new permission toggle

it hides the chat controls but not the functions / tools valves from all users

**Reply by Classic298** [reply]
@mkagit then do not use UserValves. Instead only use normal Valves.

If you add valves as uservalves you actively mean for them to be accessible to users.

Plus: Functions and Valves are not "chat controls"

**Comment by basjaburg** [comment]
So how do I get rid of user valves? To my knowledge I did not Add them

**Comment by Classic298** [comment]
@basjaburg you added a Tool or Function which has UserValves. Remove them or rename to Valves. Then they will be only available to admins from the workspace or admin panel respectively

**Comment by basjaburg** [comment]
Hi Thanks. Well that is the case: There are (to my knowledge) no UserValves.


See for yourself:


"""

parameters:
  type: object
  properties:
    chatInput:
      type: string
      title: Chat Input
      description: ''
  required:
    - chatInput
"""

from typing import Optional, Callable, Awaitable, Tuple
from pydantic import BaseModel, Field
import time
import requests
import json


def extract_event_info(event_emitter) -> Tuple[Optional[str], Optional[str]]:
    if not event_emitter or not getattr(event_emitter, "__closure__", None):
        return None, None
    for cell in event_emitter.__closure__:
        contents = cell.cell_contents
        if isinstance(contents, dict):
            return contents.get("chat_id"), contents.get("message_id")
    return None, None


class Pipe:
    class Valves(BaseModel):
        n8n_url: str = Field(
            default="webhook",
            description="default webhhook",
        )
        n8n_bearer_token: str = Field(
            default="...",
            description="Bearer token",
        )
        input_field: str = Field(default="chatInput", description="query")
        response_field: str = Field(default="output", description="response")
        emit_interval: float = Field(
            default=2.0, description="Interval (seconds) bewtween status-emits"
        )
        enable_status_indicator: bool = Field(
            default=True, description="Turn status-indicator on/off"
        )
        enable_streaming: bool = Field(
            default=False, description="Turn streaming mode on/off"
        )

    def __init__(self):
        self.type = "pipe"
        self.id = "n8n_pipe"
        self.name = "N8N Pipe"
        self.valves = self.Valves()
        self.last_emit_time = 0.0

    async def emit_status(
        self,
        __event_emitter__: Callable[[dict], Awaitable[None]],
        level: str,
        message: str,
        done: bool,
    ):
        if not __event_emitter__ or not self.valves.enable_status_indicator:
            return
        now = time.time()
        if done or (now - self.last_emit_time >= self.valves.emit_interval):
            await __event_emitter__(
                {
                    "type": "status",
                    "data": {
                        "status": "complete" if done else "in_progress",
                        "level": level,
                        "description": message,
                        "done": done,
                    },
                }
            )
            self.last_emit_time = now

    async def pipe(
        self,
        body: dict,
        __user__: Optional[dict] = None,
        __event_emitter__: Callable[[dict], Awaitable[None]] = None,
        __event_call__: Callable[[dict], Awaitable[dict]] = None,
    ) -> Optional[str]:
        await self.emit_status(__event_emitter__, "info", "Calling webhook...", False)

        chat_id, _ = extract_event_info(__event_emitter__)
        messages = body.get("messages", [])
        if not messages:
            await self.emit_status(
                __event_emitter__, "error", "sorry", True
            )
            body.setdefault("messages", []).append(
                {"role": "assistant", "content": "sorry"}
            )
            return None

        question = messages[-1].get("content", "")

        # Bouw payload
        payload = {
            "session_id": chat_id,
            self.valves.input_field: question,
        }

        if __user__:
            payload["user"] = __user__

        headers = {
            "Authorization": f"Bearer {self.valves.n8n_bearer_token}",
            "Content-Type": "application/json",
        }

        if self.valves.enable_streaming:
            headers["Accept"] = "text/event-stream"
            payload["stream"] = True
            full_response = ""
            try:
                with requests.Session() as session:
                    with session.post(
                        self.valves.n8n_url, json=payload, headers=headers, stream=True
                    ) as resp:
                        resp.raise_for_status()
                        for chunk in resp.iter_content(chunk_size=1024):
                            if not chunk:
                                continue
                            text = chunk.decode("utf-8", errors="ignore")
                            for line in text.splitlines():
                                if line.startswith("data:"):
                                    data = line[len("data:") :].strip()
                                    if data and data != "[DONE]":
                                        try:
                                            part = json.loads(data).get("token", data)
                                        except Exception:
                                            part = data
                                        full_response += part
                                        if __event_emitter__:
                                            await __event_emitter__(
                                                {"type": "token", "data": part}
                                            )
            except Exception as e:
                error_msg = f"Error making request to : {str(e)}"
                await self.emit_status(__event_emitter__, "error", error_msg, True)
                print(f"ERROR: {error_msg}")
                body["messages"].append({"role": "assistant", "content": error_msg})
                return error_msg

            body["messages"].append({"role": "assistant", "content": full_response})
            await self.emit_status(
                __event_emitter__, "info", "Streaming complete", True
            )
            return full_response
        else:
            try:
                resp = requests.post(self.valves.n8n_url, json=payload, headers=headers)
                resp.raise_for_status()
                result = resp.json().get(self.valves.response_field)
                body["messages"].append({"role": "assistant", "content": result})
                await self.emit_status(__event_emitter__, "info", "Complete", True)
                return result
            except Exception as e:
                error_msg = f"Error making request to N8N: {str(e)}"
                await self.emit_status(__event_emitter__, "error", error_msg, True)
                print(f"ERROR: {error_msg}")
                body["messages"].append({"role": "assistant", "content": error_msg})
                return error_msg

**Comment by Classic298** [comment]
@basjaburg where do you configure  the Valves? You should be able to configure them from Admin panel > functions just fine by clicking the little gear icon on the function
